packages/tracing/test/browser/request.test.ts

```diff
@@ -87,7 +87,7 @@ describe('callbacks', () => {
     it('does not create span if shouldCreateSpan returns false', () => {
```

Comment:
Can we combine this and the `'adds sentry-trace header to fetch requests'` test below into a group of four tests testing the four combinations of `always` and `never` for both `shouldCreateSpan` and `shouldAttachHeaders`, each testing both span creation and header addition?